[breaking] uniform cli commands and flag (#1542)
arduino/arduino-cli: docs/UPGRADING.md

Here you can find a list of migration guides to handle breaking changes between releases of the CLI.

## Unreleased

### `board details` arguments change

> The `board details` command now accepts only the `-b` flag to specify the FQBN.

Reviewer: ```suggestion
The `board details` command now accepts only the `--fqbn` or `-b` flags to specify the FQBN.
```

The previously deprecated `board details <FQBN>` syntax is no longer supported.

### `board attach` arguments change

> The `board attach` command now uses `-p` and `-b` flags to respectively select the board FQBN and the port.

Reviewer: ```suggestion
The `board attach` command now uses `--port` and `-p` flags to set board port and `--board` and `-b` flags to select its FQBN.
```

Here you can find a list of migration guides to handle breaking changes between releases of the CLI.

## Unreleased

### `board details` arguments change

The `board details` command now accept only the `-b` flag to specify the fqbn.

`board details <fqbn>` has been deprecated in favour of `board details -b <fqbn>`.

### `board attach` arguments change

> The `board attach` command now accept `-p` and `-b` flags to respectively select the board fqbn and the port.

Reviewer 2: ```suggestion
The `board attach` command now uses `-p` and `-b` flags to respectively select the board FQBN and the port.
```

I think "uses" makes it more clear that this is a hard change rather than an additional syntax.

> `board attach [<port|fqbn>] <sketch-path> has been deprecated in favour of`board attach [-b <fqbn> | -p <port>]f
> <sketh-path>`.

Reviewer 2: ```suggestion
The previous syntax `board attach <port>|<FQBN> [sketchPath]` is no longer supported.
```

> ### `--timeout` flag in `board list` command have been replaced by `--discovery-timeout`

Reviewer 2: ```suggestion
### `--timeout` flag in `board list` command has been replaced by `--discovery-timeout`
```
Minor grammar correction

> The flag `--timeout` in the `board list` command has been deprecated in favour of `--discovery-timeout`.

Reviewer 2: ```suggestion
The flag `--timeout` in the `board list` command is no longer supported.
```
